hibernate-core/src/main/java/org/hibernate/cfg/AvailableSettings.java

```diff
@@ -208,6 +208,185 @@
 	String CDI_BEAN_MANAGER = "javax.persistence.bean.manager";
 
 
+	// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+	// Jakarta JPA defined settings
```

Comment:
Bear in mind that there might be a need to eventually support both APIs from the same core, at least in terms of property handling, so such code might be necessary. For example, the Jakarta-transformed artifacts might still need to integrate with lagging frameworks wich have migrated only partially - unfortunately we need to allow migration in steps.

But yes these aspect are tricky. I was hoping we could at least prevent people to specify both properties by throwing an error, but there's also the case (which I see @beikov handled) in which we have to produce properties for others to read - clearly we need both in such cases, and then differentiating user error from intentb becomes impossible.
Maybe we should check at least for when both are specificed AND and being set inconsistentently.

I suggest we start by how Christian started and refine later, the support for Jakarta is going to be flagged experimental so to allow the "migrations in steps", and as other stacks consolidate the move we make ORM less lenient.

Reply:
> unfortunately we need to allow migration in steps

I could understand moving to new properties first, and then to a new API, but since we're breaking all APIs right now, I really don't see how it helps.

I suppose I don't have the big picture.

> I suggest we start by how Christian started and refine later, the support for Jakarta is going to be flagged experimental so to allow the "migrations in steps", and as other stacks consolidate the move we make ORM less lenient.

There's a history in Hibernate projects of "temporary" leniency in API and configuration properties becoming permanent. Either because we move on, or because we want to be nice to some users with specific use cases.

I would be surprised to see this evolve differently.

But as you wish...

Reply:
Fair point, yes we have such history. Let's try to improve on it but I believe the way to improve is to actually follow through on cleanups and making things stricter rather than preventing - we do need to integrate with in-progress stacks at this point.

Historically the lack of cleanup is typically been a problem of bandwith: we need to make things "strict" before declaring the feature stable, and we typically get our hand forced on timelines. We don't currently have a deadline for releasing Jakarta support as "stable" so I'm a bit more optimistic on our chances this time.